### PR TITLE
 🐛 Update ExcludeNodeDrainingAnnotation

### LINKED ISSUE #1930: ExcludeNodeDrainingAnnotation contains io.io
\bug
ExcludeNodeDrainingAnnotation = "machine.cluster.x-k8s.io.io/exclude-node-draining"
This annotation has repeated string io.io

### PR DESCRIPTION
**Pull request by prankul88**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1930

### CODE REVIEW

**Comment by prankul88** [comment]
@vincepri Can you help me in this? Thanks

File: api/v1alpha2/conversion.go [hunk 1]
```diff
@@ -120,6 +122,8 @@ func (dst *Machine) ConvertFrom(srcRaw conversion.Hub) error {
 		return err
 	}
 
+	v1alpha3.ExcludeNodeDrainingAnnotation = "machine.cluster.x-k8s.io.io/exclude-node-draining"
```
Comment (ncdc) [inline api/v1alpha2/conversion.go]: You can't change the value of a constant, nor do we want to 😄. Instead, please add logic that does the following:

- If converting from v1alpha2 to v1alpha3, if there is an annotation in the v1alpha2 machine with the key `machine.cluster.x-k8s.io.io/exclude-node-draining`, please change it to `machine.cluster.x-k8s.io/exclude-node-draining` and delete the other key
- If converting from v1alpha3 to v1alpha2, if there is an annotation in the v1alpha3 machine with the key `machine.cluster.x-k8s.io/exclude-node-draining`, please change it to `machine.cluster.x-k8s.io.io/exclude-node-draining` and delete the other key

right @vincepri?

Comment (detiber) [reply api/v1alpha2/conversion.go]: Is there an issue with changing the constant in the v1alpha3 version of the types, and using the respective constants (v1alpha2 vs v1alpha3 types) to do the comparisons in the conversion?

Comment (vincepri) [reply api/v1alpha2/conversion.go]: I see the constant has been changed in the v1alpha3 file, we only need to convert the annotation in `ConvertTo` and `ConvertFrom` methods.

Converting from v1alpha3:
- If the `v1alpha3. ExcludeNodeDrainingAnnotation` is present on the v1alpha3 object in the `Annotations` map, change it to `(v1alpha2.)ExcludeNodeDrainingAnnotation`.

Converting to v1alpha3:
- If the `(v1alpha2.)ExcludeNodeDrainingAnnotation` is present on the v1alpha2 object in the `Annotations` map, change it to `v1alpha3.ExcludeNodeDrainingAnnotation`.

Comment (prankul88) [reply api/v1alpha2/conversion.go]: Updated. PTAL! Thanks

File: api/v1alpha2/conversion.go [hunk 2]
```diff
@@ -90,6 +90,12 @@ func (src *Machine) ConvertTo(dstRaw conversion.Hub) error {
 		return err
 	}
 
+	// Manually convert ExcludeNodeDrainingAnnotation annotation if set.
+	if val, ok := src.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation]; ok {
+		src.ObjectMeta.Annotations[v1alpha3.ExcludeNodeDrainingAnnotation] = val
```
Comment (vincepri) [inline api/v1alpha2/conversion.go]:
```suggestion
		src.Annotations[v1alpha3.ExcludeNodeDrainingAnnotation] = val
```

File: api/v1alpha2/conversion.go [hunk 3]
```diff
@@ -120,6 +126,12 @@ func (dst *Machine) ConvertFrom(srcRaw conversion.Hub) error {
 		return err
 	}
 
+	// Manually convert ExcludeNodeDrainingAnnotation annotation if set.
+	if val, ok := src.ObjectMeta.Annotations[v1alpha3.ExcludeNodeDrainingAnnotation]; ok {
+		src.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation] = val
```
Comment (vincepri) [inline api/v1alpha2/conversion.go]:
```suggestion
		src.Annotations[ExcludeNodeDrainingAnnotation] = val
```

File: api/v1alpha2/conversion.go [hunk 4]
```diff
@@ -90,6 +90,12 @@ func (src *Machine) ConvertTo(dstRaw conversion.Hub) error {
 		return err
 	}
 
+	// Manually convert ExcludeNodeDrainingAnnotation annotation if set.
+	if val, ok := src.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation]; ok {
+		src.ObjectMeta.Annotations[v1alpha3.ExcludeNodeDrainingAnnotation] = val
+		delete(src.ObjectMeta.Annotations, ExcludeNodeDrainingAnnotation)
```
Comment (vincepri) [inline api/v1alpha2/conversion.go]:
```suggestion
		delete(src.Annotations, ExcludeNodeDrainingAnnotation)
```

File: api/v1alpha2/conversion.go [hunk 5]
```diff
@@ -120,6 +126,12 @@ func (dst *Machine) ConvertFrom(srcRaw conversion.Hub) error {
 		return err
 	}
 
+	// Manually convert ExcludeNodeDrainingAnnotation annotation if set.
+	if val, ok := src.ObjectMeta.Annotations[v1alpha3.ExcludeNodeDrainingAnnotation]; ok {
+		src.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation] = val
+		delete(src.ObjectMeta.Annotations, v1alpha3.ExcludeNodeDrainingAnnotation)
```
Comment (vincepri) [inline api/v1alpha2/conversion.go]:
```suggestion
		delete(src.Annotations, v1alpha3.ExcludeNodeDrainingAnnotation)
```

**Comment by vincepri** [comment]
Changes looks good

**Comment by prankul88** [comment]
@vincepri Updated.

File: api/v1alpha2/conversion.go [hunk 6]
```diff
@@ -90,6 +90,12 @@ func (src *Machine) ConvertTo(dstRaw conversion.Hub) error {
 		return err
 	}
 
+	// Manually convert ExcludeNodeDrainingAnnotation annotation if set.
+	if val, ok := src.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation]; ok {
```
Comment (detiber) [inline api/v1alpha2/conversion.go]:
```suggestion
	if val, ok := src.Annotations[ExcludeNodeDrainingAnnotation]; ok {
```

File: api/v1alpha2/conversion.go [hunk 7]
```diff
@@ -120,6 +126,12 @@ func (dst *Machine) ConvertFrom(srcRaw conversion.Hub) error {
 		return err
 	}
 
+	// Manually convert ExcludeNodeDrainingAnnotation annotation if set.
+	if val, ok := src.ObjectMeta.Annotations[v1alpha3.ExcludeNodeDrainingAnnotation]; ok {
```
Comment (detiber) [inline api/v1alpha2/conversion.go]:
```suggestion
	if val, ok := src.Annotations[v1alpha3.ExcludeNodeDrainingAnnotation]; ok {
```

**Comment by prankul88** [comment]
@detiber Done. PTAL

**Comment by detiber** [comment]
Thanks!
/lgtm

**Pull request merged**